wp-stellate.php

```diff
@@ -445,17 +448,18 @@ function stellate_encode_ids(array $ids, string $type_prefix)
       case 'purged_types':
         /** Handle types where all entities should by purged. */
         foreach ($value as $type) {
-          $selection_set .= "purge{$type}(soft: \$soft)\n";
+          $uppercase_type = ucfirst($type);
+          $selection_set .= "purge{$uppercase_type}: _purgeType(type: \"{$uppercase_type}\", soft: \$soft)\n";
         }
         break;
       default:
         if (count($value) > 0) {
           /** Handle purging individual entities by their id. */
           $uppercase_key = ucfirst($key);
           $variable_name = "\${$key}Ids";
-          $variable_definitions .= " {$variable_name}: [ID!]";
-          $selection_set .= "purge{$uppercase_key}ById: purge{$uppercase_key}(soft: \$soft, id: {$variable_name})\n";
-          $variable_values[$variable_name] = stellate_encode_ids($value, $GLOBALS['gcdn_id_prefix_map'][$key]);
+          $variable_definitions .= " {$variable_name}: [KeyFieldInput!]";
+          $selection_set .= "purge{$uppercase_key}ById: _purgeType(type: \"{$uppercase_key}\", soft: \$soft, keyFields: {$variable_name})\n";
+          $variable_values[$variable_name] = stellate_encode_ids_as_key_field_inputs($key, $value);
```

Comment:
Here we're using the string as key here that includes the `$`, meaning we'd send `{"$postIds":[...]}` as variables to the purging API which seems wrong 🤔 the thing is that it was like that before, so I'm wondering if this was ever working 🤔🤔🤔

Reply:
Interesting, I'll check this out. I only verified with our UI and it said it did purge _types_ but not sure it did actually purge them by their _ID_